pageserver/src/layered_repository/storage_layer.rs

```diff
@@ -82,6 +83,21 @@ pub struct PageReconstructData {
     pub page_img: Option<Bytes>,
 }
 
+/// Return value from Layer::get_page_reconstruct_data
+pub enum PageReconstructResult {
+    /// Got all the data needed to reconstruct the requested page
+    Complete,
+    /// This layer didn't contain all the required data, the caller should collect
+    /// more data from the returned predecessor layer at the returned LSN.
+    Continue(Lsn, Arc<dyn Layer>),
+    /// This layer didn't contain data needed to reconstruct the page version at
+    /// the returned LSN. This is usually considered an error, but might be OK
+    /// in some circumstances.
+    Missing(Lsn),
+    /// Error happened while fetching the data
+    Error(anyhow::Error),
```

Comment:
I think we should just wrap this struct in `anyhow::Result`

Reply:
+1 on that, this will save some code on using '?'

Reply:
Agreed, not being able to use '?' adds a lot of noise. Changed it that way.

There's a feature in rust nightly to do this in a more nice way: https://rust-lang.github.io/rfcs/1859-try-trait.html. But this will do for now.